src/inet/UDPEndPointImplOpenThread.h

```diff
@@ -32,6 +32,8 @@
 namespace chip {
 namespace Inet {
 
+extern otInstance * globalOtInstance;
```

Comment:
Why do we need this global?  Why can't the iterator be initialized with a pointer to the right thing?

Reply:
Because of where it is use. Currently the only layer having access to the OtInstance is the DeviceLayer in the Platform lib. Passing a pointer would require every caller to have a direct dependency to the Platform lib or to store a copy of the pointer (which would lead to how and when to initialized it). Having it global within the Inet layer is the "cleanest" way to prevent additional inter-layers dependencies given the fact that we have access to it in the UDPEndpoint Init.

Reply:
UDPEndPointImplOT is a singleton for now as we don't have any use case for multiple UDPEndpoints. It is linked to the fact that the native OpenThread implementation has only one networkInterface hence a single OTInstance. Should someone add the capability of having multiples interface, thus adding multiples OTInstance and Multiple Endpoints, this will have to change.